stripe/src/resources/payout.rs

```diff
@@ -16,20 +19,175 @@ pub struct Payout {
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct PayoutParams {
+    pub amount: u64,
+    pub currency: Currency,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub destination: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub metadata: Option<Metadata>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub method: Option<PayoutMethod>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub source_type: Option<PayoutSourceType>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub statement_descriptor: Option<String>,
+}
+
+/// The set of parameters that can be used when listing payouts.
+///
+/// For more details see [https://stripe.com/docs/api/payouts/list](https://stripe.com/docs/api/payouts/list)
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct PayoutListParams {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub arrival_date: Option<RangeQuery<Timestamp>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub created: Option<RangeQuery<Timestamp>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub destination: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ending_before: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub limit: Option<i64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub starting_after: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub status: Option<PayoutStatus>,
+}
+
+impl Payout {
+    /// Creates a new payout.
+    ///
+    /// For more details see [https://stripe.com/docs/api/payouts/create](https://stripe.com/docs/api/payouts/create).
+    pub fn create(client: &Client, params: PayoutParams) -> Result<Payout, Error> {
+        client.post("/payouts", params)
+    }
+
+    /// Retrieves the details of a payout.
+    ///
+    /// For more details see [https://stripe.com/docs/api/payouts/retrieve](https://stripe.com/docs/api/payouts/retrieve).
+    pub fn retrieve(client: &Client, payout_id: &str) -> Result<Payout, Error> {
+        client.get(&format!("/payouts/{}", payout_id))
+    }
+
+    /// Updates a payout's properties.
+    ///
+    /// For more details see [https://stripe.com/docs/api/payouts/update](https://stripe.com/docs/api/payouts/update).
+    pub fn update(
+        client: &Client,
+        payout_id: &str,
+        metadata: Option<Metadata>,
+    ) -> Result<Payout, Error> {
+        client.post(&format!("/payouts/{}", payout_id), metadata)
+    }
+
+    /// List all payouts.
+    ///
+    /// For more details see [https://stripe.com/docs/api/payouts/list](https://stripe.com/docs/api/payouts/list).
+    pub fn list(client: &Client, params: PayoutListParams) -> Result<Vec<Payout>, Error> {
```

Comment:
The return type for this should probably be `List<Payout>` (the result of Stripe's APIs is not a flat array; the `List<T>` type implements pagination via the Stripe API.